### PR TITLE
Project abstraction

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/StandardTigrisClient.java [hunk 1]
```diff
@@ -71,71 +58,9 @@ public static StandardTigrisClient getInstance(TigrisConfiguration tigrisConfigu
   }
 
   @Override
-  public TigrisDatabase getDatabase(String databaseName) {
+  public TigrisDatabase getDatabase(String projectName) {
```
Comment (ovaistariq) [inline client/src/main/java/com/tigrisdata/db/client/StandardTigrisClient.java:61]: The project name should already be known at this point, so we don't need to ask the project name?

File: client/src/main/java/com/tigrisdata/db/client/TigrisAsyncClient.java [hunk 2]
```diff
@@ -14,43 +14,18 @@
...
-  CompletableFuture<TigrisAsyncDatabase> createDatabaseIfNotExists(String databaseName);
-
-  /**
-   * Drops the database
-   *
-   * @param databaseName name of the database
-   * @return a future to the {@link DropDatabaseResponse}
-   */
-  CompletableFuture<DropDatabaseResponse> dropDatabase(String databaseName);
+  TigrisAsyncDatabase getDatabase(String projectName);
```
Comment (ovaistariq) [inline client/src/main/java/com/tigrisdata/db/client/TigrisAsyncClient.java:28]: It shouldn't be asking for the project name

File: client/src/main/java/com/tigrisdata/db/client/TigrisClient.java [hunk 3]
```diff
@@ -16,45 +16,17 @@
...
-
-  /**
-   * Drops the database
-   *
-   * @param databaseName name of the database
-   * @return an instance of {@link DropDatabaseResponse} from server
-   * @throws TigrisException in case of auth error or any other failure.
-   */
-  DropDatabaseResponse dropDatabase(String databaseName) throws TigrisException;
+  TigrisDatabase getDatabase(String projectName);
```
Comment (ovaistariq) [inline client/src/main/java/com/tigrisdata/db/client/TigrisClient.java:29]: Same as above

Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/TigrisClient.java:29]: Do you need to pass `projectName` as a param here?

File: client/src/main/java/com/tigrisdata/db/client/StandardTigrisClient.java (hunk 1 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/StandardTigrisClient.java]: fixed.

File: client/src/main/java/com/tigrisdata/db/client/TigrisClient.java (hunk 3 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/TigrisClient.java]: updated.

**Review by himank — APPROVED** [review]

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**